specs/0121-translation-memory-management.md

```diff
@@ -0,0 +1,43 @@
+- Feature Name: Translation Memory Management
+- Created: 2024-08-28
+- Associated Issue: #3031
+
+# Summary
+
+Add ability to edit, delete and upload Translation Memory (TM) entries.
+
+# Motivation
+
+Translation memory is a fundamental tool for faster and more consistent translations, and has become even more important with pretranslation, since it's used to train custom machine translation engines in Google AutoML Translation.
+
+Support for TM in Pontoon is limited:
+* Localizers cannot edit or delete existing TM entries directly.
+* TM entries can be changed indirectly via translation workspace, but only if the corresponding strings are still available for translation.
+* Localizers cannot upload their own translation memories.
+
+# Feature explanation
+
+A new "Translation Memory" tab is added to the Team pages, which lists all TM entries for the team. For each entry, source string and translation are provided, clicking on which leads to the translation workspace for the corresponding string. Entries are listed in alphabetic order.
```

Comment:
Why not use the existing translation view for this? Why not list "Translation Memory" among the projects for the Team, and use a slightly modified translate view for it?

Reply:
I was thinking about this initially, because it would match the behavior of https://pontoon.mozilla.org/projects/terminology/. We could even store uploaded .tmx files as files for that project.

But I worry about the backend complexity and speed. Unlike Terminology, TM entries are constantly added, so we'd need to sync between the actual TM and the TM Project a lot. The fact that original strings in Terminology are the same for all locales, and different for each locale for TM also doesn't help.

Reply:
To clarify, I was not suggesting having a separate TM Project, but to write the changes into TM the same way that they would be written in the current proposal.

Using the same front-end as normal projects shouldn't mean that the back-end needs to also match.